timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStoreTest.java

```diff
@@ -315,61 +249,161 @@ public void canGetAgreedStateAfterNodeDown() {
+            batchPaxosAcceptors.add(ToggleableExceptionProxy.newProxyInstance(
+                    BatchPaxosAcceptor.class, batchAcceptor, failureController, EXCEPTION));
+
+            BatchPaxosLearner batchLearner = new LocalBatchPaxosLearner(components);
+            batchPaxosLearners.add(ToggleableExceptionProxy.newProxyInstance(
+                    BatchPaxosLearner.class, batchLearner, failureController, EXCEPTION));
+        }
+
+        if (useBatch) {
+            AutobatchingPaxosAcceptorNetworkClientFactory acceptorNetworkClientFactory =
+                    AutobatchingPaxosAcceptorNetworkClientFactory.create(
+                            batchPaxosAcceptors,
+                            KeyedStream.of(batchPaxosAcceptors.stream())
+                                    .map($ -> new CheckedRejectionExecutorService(executor))
+                                    .collectToMap(),
+                            QUORUM_SIZE);
+            acceptorClient = acceptorNetworkClientFactory.paxosAcceptorForClient(CLIENT);
+
+            List<AutobatchingPaxosLearnerNetworkClientFactory> learnerNetworkClientFactories =
+                    batchPaxosLearners.stream()
+                            .map(localLearner -> LocalAndRemotes.of(
+                                    localLearner,
+                                    batchPaxosLearners.stream()
+                                            .filter(remoteLearners -> remoteLearners != localLearner)
+                                            .collect(toList())))
+                            .map(localAndRemotes -> AutobatchingPaxosLearnerNetworkClientFactory.createForTests(
+                                    localAndRemotes, executor, QUORUM_SIZE))
+                            .collect(toList());
+
+            learnerClientsByNode = learnerNetworkClientFactories.stream()
+                    .map(factory -> factory.paxosLearnerForClient(CLIENT))
+                    .collect(toList());
+
+            closer.register(acceptorNetworkClientFactory);
+            learnerNetworkClientFactories.forEach(closer::register);
+        } else {
+            acceptorClient = SingleLeaderAcceptorNetworkClient.createLegacy(
+                    acceptors,
+                    QUORUM_SIZE,
+                    Maps.toMap(acceptors, $ -> executor),
+                    PaxosConstants.CANCEL_REMAINING_CALLS);
+
+            learnerClientsByNode = learners.stream()
+                    .map(learner -> SingleLeaderLearnerNetworkClient.createLegacy(
+                            learner,
+                            learners.stream()
+                                    .filter(otherLearners -> otherLearners != learner)
+                                    .collect(toList()),
+                            QUORUM_SIZE,
+                            Maps.toMap(learners, $ -> executor),
+                            PaxosConstants.CANCEL_REMAINING_CALLS))
+                    .collect(toList());
+        }
```

Comment:
I know this was written before your PR, but I'd probably take the chance to better split these into smaller methods:

So it ends up looking something like:

```
if (useBatch) {
   setUpSomethingSomethingUsingBatch(...);
} else {
   setUpSomethingSomethingWithoutBatch(...);
}

```